File: src/fix_static.php
<?php
$directory = '/var/www/html/public/static';
$oldText = "        try {\n            const body = { test_id: EXAM_CONFIG.testId };\n            if (CSRF_NAME) body[CSRF_NAME] = CSRF_HASH;";
$newText = "        try {\n            // Because this is a static page, we must fetch the CSRF token first\n            try {\n                const csrfRes = await fetch(API + '/health', { credentials: 'same-origin' });\n                const csrfToken = csrfRes.headers.get('X-CSRF-TOKEN');\n                if (csrfToken) {\n                    CSRF_HASH = csrfToken;\n                    $.ajaxSetup({ headers: { 'X-CSRF-TOKEN': csrfToken } });\n                }\n            } catch(e) {}\n\n            const body = { test_id: EXAM_CONFIG.testId };\n            if (CSRF_NAME) body[CSRF_NAME] = CSRF_HASH;";

$iterator = new RecursiveIteratorIterator(new RecursiveDirectoryIterator($directory));
foreach ($iterator as $file) {
    if ($file->isFile() && $file->getExtension() === 'html') {
        $filepath = $file->getPathname();
        $content = file_get_contents($filepath);
        if (strpos($content, $oldText) !== false) {
            $content = str_replace($oldText, $newText, $content);
            file_put_contents($filepath, $content);
            echo "Updated {$filepath}\n";
        }
    }
}

File: fix_static.py
import os

directory = 'src/public/static'
old_text = """        try {
            const body = { test_id: EXAM_CONFIG.testId };
            if (CSRF_NAME) body[CSRF_NAME] = CSRF_HASH;"""

new_text = """        try {
            try {
                const csrfRes = await fetch(API + '/health', { credentials: 'same-origin' });
                const csrfToken = csrfRes.headers.get('X-CSRF-TOKEN');
                if (csrfToken) {
                    CSRF_HASH = csrfToken;
                    $.ajaxSetup({ headers: { 'X-CSRF-TOKEN': csrfToken } });
                }
            } catch(e) {}
            const body = { test_id: EXAM_CONFIG.testId };
            if (CSRF_NAME) body[CSRF_NAME] = CSRF_HASH;"""

for root, dirs, files in os.walk(directory):
    for file in files:
        if file.endswith('.html'):
            filepath = os.path.join(root, file)
            with open(filepath, 'r', encoding='utf-8') as f:
                content = f.read()
            
            if old_text in content:
                content = content.replace(old_text, new_text)
                with open(filepath, 'w', encoding='utf-8') as f:
                    f.write(content)
                print(f"Updated {filepath}")
